kit/Kit.cpp

```diff
@@ -2441,17 +2444,39 @@ void Document::drainQueue()
             }
         }
 
+        const size_t tilesQueued = _queue->getTileQueueSize();
         if (processInputEnabled() && !isLoadOngoing() &&
-            !isBackgroundSaveProcess() && _queue->getTileQueueSize() > 0)
+            !isBackgroundSaveProcess() && tilesQueued > 0)
         {
+            // Render all tiles if idle.
```

Comment:
All pending tiles could be a huge queue of work that could take perhaps seconds to do - if we've done a big scroll. Why would we choose to block all users for seconds when they may have more immediate things they want doing in order to render all the work.

This approach seems to be essentially back where we were before we last discussed this at some length.